ocppj/ocppj.go

```diff
@@ -198,10 +198,6 @@ const (
 	FormatViolationV16            ocpp.ErrorCode = "FormationViolation"            // Payload for Action is syntactically incorrect or not conform the PDU structure for Action. This is only valid for OCPP 1.6
 )
 
-var (
-	FormationViolation = FormatViolationV16 // Used as constant, but can be overwritten depending on protocol version. Sett FormatViolationV16 and FormatViolationV2.
-)
-
 func IsErrorCodeValid(fl validator.FieldLevel) bool {
```

Comment:
By removing the global `FormationViolation` variable this validation function will not work as expected, since the switch case below checks for this value.

I see two options:
1. move the validation function to the 1.6/2.0 packages respectively
  - cleaner but will not allow using 1.6+2.0 within the same process (something which is apprently a requirement for some devs)
2. replace `FormationViolation` in the switch case with `FormatViolationV16, FormatViolationV2`
  - works with both proto versions at the same time but doesn't prevent other implementations from sending an "incorrect" value for a specific version

I'd honestly go with 2, cause it's easier and will make everyone happy in the end.

Reply:
> By removing the global FormationViolation variable this validation function will not work as expected, since the switch case below checks for this value.

Doh: I see. It could work if the validation could get access to the endpoint- but I guess that's not something that the validator would allow hooking in (something like a validation context)?

> move the validation function to the 1.6/2.0 packages respectively

It would not help either since clients and servers would still race.

> replace FormationViolation in the switch case with FormatViolationV16, FormatViolationV2

I can't judge but it seems nice and simple and still better than a data race?